plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -1651,6 +1656,124 @@ public void testExternalLocationWithTrailingSpace()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test(dataProvider = "s3SelectFileFormats")
+    public void testS3SelectPushdown(String tableProperties)
+    {
+        Session usingAppendInserts = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "insert_existing_partitions_behavior", "APPEND")
+                .build();
+        List<String> values = ImmutableList.of(
```

Comment:
Would you please include empty string values?

Reply:
S3 Select is treating CSV column values as String. Unless there is trimming done on each entry of the list, the column values will contain spaces - e.g.
If this line is sent as is to Select:
```
1, true, 11, 111, 1111, 11111, 'one', 1.1, DATE '2020-01-01'
```
then `id='1'`, `bool_t=' true'`, `tiny_t=' 11'` etc.

You can find a sample file for the expected format here: https://github.com/trinodb/trino/blob/master/plugin/trino-hive-hadoop2/conf/files/test_table_csv_scan_range_select_pushdown_1.csv

Reply:
I'm not really following this one, can you elaborate? I tried testing out adding whitespace around the values of a textfile/csv but it completely fails to parse in Trino, regardless of if S3 select is enabled or not.

Reply:
A valid input for Select is:
```
id,bool_t,tiny_t,small_t,int_t,big_t,string_t,decimal_t
1,true,11,111,1111,11111,one,1.1
2,true,22,222,2222,22222,two,2.2
3,NULL,NULL,NULL,NULL,NULL,NULL,NULL
4,false,44,444,4444,44444,'four',
```
In this case for example: `id='1', bool_t='true', tiny_t='11'` - quotes are used here to emphasize string values.

What I was trying to say is that from a high-level look the list seems like the input data would look like:
```
1, true, 11, 111, 1111, 11111, 'one', 1.1, DATE '2020-01-01'
2, true, 22, 222, 2222, 22222, 'two', 2.2, DATE '2020-02-02'
3, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL
4, false, 44, 444, 4444, 44444, 'four', 4.4, DATE '2020-04-04'
```
This is supported on Select from a parsing perspective, but would change the values:
`id='1'`, `bool_t=' true'` (meaning '<space>true'), `tiny_t=' 11'` (meaning '<space>11').

I didn't work on Trino codebase in a while, so if the data gets without the whitespaces in Trino/Select, then that's fine 👍